pilot/pkg/proxy/envoy/v2/debug.go

```diff
@@ -319,53 +290,99 @@ func (s *DiscoveryServer) authenticationz(w http.ResponseWriter, req *http.Reque
 		}
 	}
 	mostRecentProxy = connections[mostRecent].modelNode
-	_, _ = fmt.Fprintf(w, "\n[\n")
 	svc, _ := s.Env.ServiceDiscovery.Services()
+	info := []*AuthenticationDebug{}
 	for _, ss := range svc {
+		if ss.MeshExternal {
+			// Skip external services
+			continue
+		}
 		for _, p := range ss.Ports {
-			info := AuthenticationDebug{
-				Host: string(ss.Hostname),
-				Port: p.Port,
-			}
-			// TODO: support querying with labels.
-			authnConfig := s.Env.IstioConfigStore.AuthenticationPolicyForWorkload(ss, nil, p)
-			info.AuthenticationPolicyName = configName(authnConfig)
-			var serverProtocol, clientProtocol authProtocol
-			if authnConfig != nil {
-				policy := authnConfig.Spec.(*authn.Policy)
-				mtls := authn_alpha1.GetMutualTLS(policy)
-				serverProtocol = getServerAuthProtocol(mtls)
-			} else {
-				serverProtocol = getServerAuthProtocol(nil)
-			}
-			info.ServerProtocol = authProtocolToString(serverProtocol)
-
+			authnPolicy := s.globalPushContext().AuthenticationPolicyForWorkload(ss, p)
 			destConfig := s.globalPushContext().DestinationRule(mostRecentProxy, ss)
-			info.DestinationRuleName = configName(destConfig)
-			if destConfig != nil {
-				rule := destConfig.Spec.(*networking.DestinationRule)
-				clientProtocol = clientAuthProtocol(rule, p)
-			} else {
-				clientProtocol = authnHTTP
-			}
-			info.ClientProtocol = authProtocolToString(clientProtocol)
+			info = append(info, AnalyzeMTLSSettings(ss.Hostname, p, authnPolicy, destConfig)...)
+		}
+	}
 
-			if (clientProtocol & serverProtocol) == 0 {
-				if clientProtocol == authnCustomMTls && serverProtocol != authnHTTP {
-					info.TLSConflictStatus = "MAY CONFLICT"
-				} else {
-					info.TLSConflictStatus = "CONFLICT"
-				}
-			} else {
-				info.TLSConflictStatus = "OK"
-			}
-			if b, err := json.MarshalIndent(info, "  ", "  "); err == nil {
-				_, _ = w.Write(b)
-			}
-			_, _ = fmt.Fprintf(w, ",\n")
+	if b, err := json.MarshalIndent(info, "  ", "  "); err == nil {
+		_, _ = w.Write(b)
+	}
+}
+
+// AnalyzeMTLSSettings returns mTLS compatibility status between client and server policies.
+func AnalyzeMTLSSettings(hostname host.Name, port *model.Port, authnPolicy *authn.Policy,
+	destConfig *model.Config) []*AuthenticationDebug {
+	// TODO(diemvu): add policy config name to the cache push config for this.
+	authnPolicyName := "-"
```

Comment:
I'm confused by these. policy name is `???` if the authnPolicy is not nil, and `-` if it's nil?

Reply:
Also could you add some comment, what does `-` stands for? we use the same `-` in the client below as well.

Reply:
`-` means no config (policy, rule). `???` is just a temporary for now, as I need to change the cache push context to propagate this information (CRD name) for this debugging tool.

Reply:
Use constants for `-` as something `UNSET`, and comments for `???` then.

For now, they look like magic.

Reply:
done